client/dashboard/sites/scan/components/ignore-threat-modal.tsx

```diff
@@ -0,0 +1,82 @@
+import { ignoreThreatMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import {
+	Button,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	ExternalLink,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { ButtonStack } from '../../../components/button-stack';
+import { Notice } from '../../../components/notice';
+import { CODEABLE_JETPACK_SCAN_URL } from '../constants';
+import { ThreatDescription } from './threat-description';
+import { ThreatsDetailCard } from './threats-detail-card';
+import type { Threat } from '@automattic/api-core';
+import type { RenderModalProps } from '@wordpress/dataviews';
+
+interface IgnoreThreatModalProps extends RenderModalProps< Threat > {
+	siteId: number;
+}
+
+export function IgnoreThreatModal( {
+	items,
+	closeModal,
+	onActionPerformed,
+	siteId,
+}: IgnoreThreatModalProps ) {
+	const threat = items[ 0 ];
+
+	const ignoreThreat = useMutation( ignoreThreatMutation( siteId ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const handleIgnoreThreat = () => {
+		ignoreThreat.mutate( threat.id, {
+			onSuccess: () => {
+				closeModal?.();
+				onActionPerformed?.( [ threat ] );
+				createSuccessNotice( __( 'Threat ignored.' ), { type: 'snackbar' } );
+			},
+			onError: () => {
+				closeModal?.();
+				createErrorNotice( __( 'Failed to ignore threat. Please try again.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	return (
+		<VStack spacing={ 4 }>
+			<Text variant="muted">{ __( 'Jetpack will be ignoring the following threat:' ) }</Text>
+			<ThreatsDetailCard threats={ [ threat ] } />
+			<ThreatDescription threat={ threat } />
+			<Notice variant="error">
+				{ createInterpolateElement(
+					__(
+						'By ignoring this threat you confirm that you have reviewed the detected code and assume the risks of keeping a potentially malicious file on your site. If you are unsure please request an estimate with <codeable />.'
+					),
+					{
+						codeable: <ExternalLink href={ CODEABLE_JETPACK_SCAN_URL }>Codeable</ExternalLink>,
+					}
+				) }
+			</Notice>
+			<ButtonStack justify="flex-end">
+				<Button variant="tertiary" onClick={ closeModal }>
+					{ __( 'Cancel' ) }
+				</Button>
+				<Button
+					variant="primary"
+					onClick={ handleIgnoreThreat }
+					isBusy={ ignoreThreat.isPending }
```

Comment:
Is this button expected to stay blue when busy? Or should it turn gray like the `Scan now`?

<img width="218" height="99" alt="image" src="https://github.com/user-attachments/assets/551cf311-c9ae-4926-b03e-64d7b23d6a35" />

Reply:
Yes, the `Scan now` uses a different variant (`secondary`), which makes it grey when busy + disabled:

<img width="250" height="98" alt="CleanShot 2025-09-19 at 22 03 21@2x" src="https://github.com/user-attachments/assets/0a64a87c-c53a-4bac-8995-7bbf35afcfb7" />

https://wordpress.github.io/gutenberg/?path=/story/components-button--secondary&args=variant:secondary;isBusy:!true;disabled:!true

The core button component keeps the blue when busy + disabled when the variant is `primary`:

<img width="250" height="98" alt="CleanShot 2025-09-19 at 22 04 05@2x" src="https://github.com/user-attachments/assets/7033f370-c18e-4dae-b144-d52301b86626" />